vulnerabilities/importers/apache_httpd.py

```diff
@@ -49,47 +53,58 @@ def updated_advisories(self):
         # skips processing the response further to avoid duplicate work
 
         if create_etag(data_src=self, url=self.url, etag_key="ETag"):
-            data = fetch_xml(self.url)
-            advisories = to_advisories(data)
+            links = fetch_links(self.url)
+            advisories = []
+            for link in links:
+                data = requests.get(link).json()
+                advisories.append(self.to_advisory(data))
             return self.batch_advisories(advisories)
 
         return []
 
-
-def to_advisories(data):
-    advisories = []
-    for issue in data:
+    def to_advisory(self, data):
+        cve = data["CVE_data_meta"]["ID"]
+        description = data["description"]["description_data"]
+        summary = next((item["value"] for item in description if item["lang"] == "eng"), "")
+        severity = VulnerabilitySeverity(
+            system=scoring_systems["apache_httpd"],
+            value=data["impact"][0]["other"],
```

Comment:
This feels as being magical with `[0]`... Let's try something more explicit and easier to read possibly like the suggestion above. In particular if there is no impact, then no `VulnerabilitySeverity` should be created. As it the code may be brittle and crash easily if this happens.

Reply:
```
impacts = data.get("impact", [])
impact = None
for imp in impacts:
     value = imp.get("other")
     if value is not None:
          impact = value
          break

if impact is not None:
     severity = VulnerabilitySeverity(
          system=scoring_systems["apache_httpd"],
          value=impact,
     )
     reference = Reference(reference_id=cve, url=self.url + cve + ".json", severities=[severity])
else:
     reference = Reference(reference_id=cve, url=self.url + cve + ".json")
```
please ignore concatenation of url I am working on it.